libtransmission/torrent.cc

```diff
@@ -2282,9 +2311,9 @@ void tr_torrent::set_download_dir(std::string_view path, bool is_new_torrent)
             recheck_completeness();
         }
     }
-    else if (error == TR_STAT_LOCAL_ERROR && !setLocalErrorIfFilesDisappeared(this))
+    else if (!std::empty(error_) && !setLocalErrorIfFilesDisappeared(this))
```

Comment:
Is it better to have the "files disappeared" local error overwrite the existing tracker warning/error?

If no, then this will revert the behaviour:

```suggestion
    else if (error_.error_type() != TR_STAT_LOCAL_ERROR && !setLocalErrorIfFilesDisappeared(this))
```

Reply:
restored it to the previous code in e9dd5001122679ad7a8e463c8c481e486b140380